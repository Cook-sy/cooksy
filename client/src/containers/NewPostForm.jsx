import React, { Component } from 'react';
import { Field, reduxForm } from 'redux-form';
import { connect } from 'react-redux';

class NewPostForm extends Component {
  render() {
    return (
      <form>
        <div>
          <label>Name</label>
          <div>
            <Field
              name="name"
              component="input"
              type="text"
              placeholder="Your Name"
            />
          </div>
        </div>
        <div>
          <label>Description</label>
          <div>
            <Field
              name="description"
              component="textarea"
              type="textarea"
              placeholder="Description"
            />
          </div>
        </div>
        <div>
          <label>Delivery Date</label>
          <div>
            <Field
              name="delivery-date"
              component="input"
              type="date"
            />
          </div>
        </div>
        <div>
                <label>Photos</label>
                <div>
                  <Field
                    name="photo_1"
                    component="input"
                    type="text"
                    placeholder="Your Name"
                  />
                </div>
                <div>
                  <Field
                    name="photo_2"
                    component="input"
                    type="text"
                    placeholder="Your Name"
                  />
                </div>
                <div>
                  <Field
                    name="photo_3"
                    component="input"
                    type="text"
                    placeholder="Your Name"
                  />
                </div>
                <div>
                  <Field
                    name="photo_4"
                    component="input"
                    type="text"
                    placeholder="Your Name"
                  />
                </div>
              </div>
        <div>
          <label>Price</label>
          <div>
            <Field name="price" component="input" type="text"/>
          </div>
          <label>Servings</label>
          <div>
            <Field name="servings" component="input" type="text"/>
          </div>
        </div>
        <div>
          <label>Addresee</label>
          <div>
            <Field name="address" component="input" type="text"/>
          </div>
          <label>City</label>
          <div>
            <Field name="city" component="input" type="text"/>
          </div>
          <label>State</label>
          <div>
            <Field name="state" component="input" type="text"/>
          </div>
          <label>Zipcode</label>
          <div>
            <Field name="zipcode" component="input" type="text"/>
          </div>
        </div>
        <div>
          <button type="submit" >Submit</button>
          <button type="button" >
            Clear Values
          </button>
        </div>
      </form>
    );
  }
}
       
export default reduxForm({
  form: 'NewPostForm' 
})(NewPostForm);