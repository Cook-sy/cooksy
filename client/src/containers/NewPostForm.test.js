import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import NewPostForm from './NewPostForm';
import { reduxForm } from 'redux-form';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

const store = createStore(() => ({}));
const Decorated = reduxForm({ form: 'testForm' })(NewPostForm);

describe('NewPostForm', () => {
  it('renders without crashing', () => {
    shallow(<NewPostForm />);
  });

  it('renders correctly', () => {
    const component = renderer.create(<Provider 
    	store={store}>
        <Decorated />
      </Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
