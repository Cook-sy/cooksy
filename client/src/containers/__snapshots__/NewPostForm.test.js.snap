// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPostForm renders correctly 1`] = `
<form>
  <div>
    <label>
      Name
    </label>
    <div>
      <input
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Description
    </label>
    <div>
      <textarea
        name="description"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Description"
        type="textarea"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Delivery Date
    </label>
    <div>
      <input
        name="delivery-date"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="date"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Photos
    </label>
    <div>
      <input
        name="photo_1"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        name="photo_2"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        name="photo_3"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        name="photo_4"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder="Your Name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Price
    </label>
    <div>
      <input
        name="price"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <label>
      Servings
    </label>
    <div>
      <input
        name="servings"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <label>
      Addresee
    </label>
    <div>
      <input
        name="address"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <label>
      City
    </label>
    <div>
      <input
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <label>
      State
    </label>
    <div>
      <input
        name="state"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
    <label>
      Zipcode
    </label>
    <div>
      <input
        name="zipcode"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div>
    <button
      type="submit"
    >
      Submit
    </button>
    <button
      type="button"
    >
      Clear Values
    </button>
  </div>
</form>
`;
